src/main/emscripten/SDL_sysmain_callbacks.c

```diff
@@ -24,9 +24,62 @@
 
 #include <emscripten.h>
 
+// For Emscripten, we let you use SDL_HINT_MAIN_CALLBACK_RATE, because it might be useful to drop it super-low for
+//  things like loopwave that don't really do much but wait on the audio device, but be warned that browser timers
+//  are super-unreliable in modern times, so you likely won't hit your desired callback rate with good precision.
+// Almost all apps should leave this alone, so we can use requestAnimationFrame, which is intended to run reliably
+//  at the refresh rate of the user's display.
+static Uint32 callback_rate_increment = 0;
+static bool iterate_after_waitevent = false;
+static bool callback_rate_changed = false;
+static void SDLCALL MainCallbackRateHintChanged(void *userdata, const char *name, const char *oldValue, const char *newValue)
+{
+    callback_rate_changed = true;
+    iterate_after_waitevent = newValue && (SDL_strcmp(newValue, "waitevent") == 0);
+    if (iterate_after_waitevent) {
+        callback_rate_increment = 0;
+    } else {
+        const double callback_rate = newValue ? SDL_atof(newValue) : 0.0;
+        if (callback_rate > 0.0) {
+            callback_rate_increment = (Uint32) SDL_NS_TO_MS((double) SDL_NS_PER_SECOND / callback_rate);
+        } else {
+            callback_rate_increment = 0;
+        }
+    }
+}
+
+// just tell us when any new event is pushed on the queue, so we can check a flag for "waitevent" mode.
+static bool saw_new_event = false;
+static bool SDLCALL EmscriptenMainCallbackEventWatcher(void *userdata, SDL_Event *event)
+{
+    saw_new_event = true;
+    return true;
+}
+
 static void EmscriptenInternalMainloop(void)
 {
-    const SDL_AppResult rc = SDL_IterateMainCallbacks(true);
+    // callback rate changed? Update emscripten's mainloop iteration speed.
+    if (callback_rate_changed) {
+        SDL_Log("NEW CALLBACK RATE: %u", (unsigned int) callback_rate_increment);
```

Comment:
This is just debug, right?

Reply:
Shoot, I always forget to comment those out!